### PR TITLE
Async

### PR DESCRIPTION
**Pull request by hupebln**
# Async implemented
## Description
I've implemented async calls to speed up the calls, especially when you need to call multiple times the "cli" endpoint it saves a lot time.

### CODE REVIEW

File: napalm_arubaoss/ArubaOS.py [hunk 1]
```diff
@@ -321,42 +436,41 @@ def _commit_candidate(self, config):
...
-                item['mac'] = self._mac_reformat(entry['mac_address'])
-                item['interface'] = entry['port_id']
-                item['vlan'] = entry['vlan_id']
-                item['active'] = True
-                """ Not supported:
-                item['static'] = False
-                item['moves'] = 0
-                item['last_move'] = 0.0
-                """
+            for entry in resp.json()._get('mac_table_entry_element'):
```
Comment (gcotone) [inline napalm_arubaoss/ArubaOS.py:452]: @hupebln There are many occurrences of this: `dict.get` replaced by `dict._get`. Only `self._apisession.get` should be replaced.

File: napalm_arubaoss/ArubaOS.py [hunk 2]
```diff
@@ -321,42 +436,41 @@ def _commit_candidate(self, config):
...
-        url = self._api_url + 'ipaddresses'
         "Looks like there's a bug n ArubaOS and is not returning IPv6"
 
-        resp = self._apisession.get(url)
+        output = {}
+        resp = self._get(self._ipaddresses_url)
         if resp.status_code == 200:
-            output = {}
-            for address in resp.json().get('ip_address_subnet_element'):
+            for address in resp.json()._get('ip_address_subnet_element'):
```
Comment (gcotone) [inline napalm_arubaoss/ArubaOS.py:473]: @hupebln here again

File: napalm_arubaoss/ArubaOS.py [hunk 3]
```diff
@@ -569,8 +696,8 @@ def _backup_config(self, config='running', destination='backup'):
         else:
             "unsupported argument; raise error"
             return False
-        cmd_post = self._apisession.post(url, json=payload)
-        if not 300 > cmd_post.status_code >= 200:
+        cmd_post = self._post(url, json=payload)
+        if not cmd_post.ok:
```
Comment (gcotone) [inline napalm_arubaoss/ArubaOS.py:700]: @hupebln Isn't this only matching "200 OK"? The REST API doesn't always return 200 OK, sometimes 201 or 204

**Review by gcotone — COMMENTED** [review]
@hupebln to double check

File: napalm_arubaoss/ArubaOS.py (hunk 3 above)
Comment (hupebln) [reply napalm_arubaoss/ArubaOS.py:700]: It's True if it is less than 400. Isn't it what you tried accomplish with this statement?

Edit: I know you used 300, but that would be a forwarding for example. I can revert it, but "ok" is actually here for cases where you wanna check if the call was successful.

File: napalm_arubaoss/ArubaOS.py (hunk 1 above)
Comment (hupebln) [reply napalm_arubaoss/ArubaOS.py:452]: I know why this happened. I refactored the class methods and renamed " get" to "_get" and didn't see PyCharm was also fetching the code you mentioned.

Comment (hupebln) [reply napalm_arubaoss/ArubaOS.py:452]: I've reverted it.

File: napalm_arubaoss/ArubaOS.py (hunk 3 above)
Comment (gcotone) [reply napalm_arubaoss/ArubaOS.py:700]: I thought it was only matching 200. All good then

File: napalm_arubaoss/ArubaOS.py (hunk 1 above)
Comment (gcotone) [reply napalm_arubaoss/ArubaOS.py:452]: thanks @hupebln ! 
Ref: 3a3033ff5e7b0cc5c357228cf8dfd230c21c19d3

**Pull request merged**